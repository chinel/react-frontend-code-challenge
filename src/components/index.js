export { default as DivisionSummary } from "./DivisionSummary/DivisionSummary";
export { default as Header } from "./Header/Header";
export { default as InfoBox } from "./InfoBox/InfoBox";
export { default as ModuleHistory } from "./ModuleHistory/ModuleHistory";
export { default as TableSummary } from "/TableSummary/TableSummary";
