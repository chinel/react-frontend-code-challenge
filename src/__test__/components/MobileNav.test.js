import React from "react";
import { render } from "@testing-library/react";
import useStyles from "../../components/MobileNav/MobileNavStyles";
import { MobileNav } from "../../components";

function checkStyles() {
  let values;

  function MockedComponent() {
    values = useStyles();
    return null;
  }
  render(<MockedComponent />);
  return values;
}

describe("<MobileNav/>", () => {
  it("renders the <MobileNav/> with populated data", () => {
    const { container, queryByText } = render(
      <MobileNav toggleDrawerHandler={() => {}} status={{ status: "left" }} />
    );

    expect(queryByText("Divisions")).toBeFalsy();
    expect(queryByText("Entries")).toBeFalsy();
    expect(queryByText("Home")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("validate the return values of useStyles hook", () => {
    const validateStlyes = checkStyles();

    expect(validateStlyes.list).toBeTruthy();
  });
});
