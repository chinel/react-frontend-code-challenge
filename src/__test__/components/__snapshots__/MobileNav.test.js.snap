// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileNav/> renders the <MobileNav/> with populated data 1`] = `<div />`;
